src/main/java/com/icthh/xm/ms/entity/service/XmEntitySpecService.java

```diff
@@ -314,11 +316,32 @@ public Optional<StateSpec> findState(String key, String stateKey) {
         return getTypeSpecs().get(key).getStates().stream().filter(s -> s.getKey().equals(stateKey)).findFirst();
     }
 
+    @IgnoreLogginAspect
+    public FunctionSpec findFunctionByKey(String functionKey) {
+        return xmEntitySpecContextService.functionsByTenant(getTenantKeyValue()).get(functionKey);
+    }
+
+    public List<FunctionMetaInfo> findFunctionMetaInfoByKey(String functionKey) {
+        return xmEntitySpecContextService.functionsMetaInfoByTenant(getTenantKeyValue()).stream()
+            .filter(f -> f.functionKey().equals(functionKey)).toList();
+    }
+
+    public List<FunctionMetaInfo> findAllFunctionMetaInfo() {
```

Comment:
lets make this method nullSafe, like other methods with collection return.
In next call we are calling ```nullSafe(allFunctionMetaInfo)```